syntaxes/gdb.tmLanguage.json

```diff
@@ -156,13 +230,50 @@
 				},
 				{
 					"name": "meta.set.history.save.gdb",
-					"match": "^\\s*(set)\\s+(history)\\s+(save(?:\\s+(?:on|off))?)\\s*$",
+					"match": "^\\s*(set)\\s+(hi(?:s(?:t(?:o(?:r(?:y)?)?)?)?)?\\s+sa(?:v(?:e)?)?)\\s+(:on|off)\\s*$",
+					"captures": {
+						"1": { "name": "keyword.control.command.gdb" },
+						"2": { "name": "support.function.option.gdb" },
+						"3": { "name": "variable.other.variable.gdb" }
+					}
+				},
+				{
+					"name": "meta.set.history.size.gdb",
+					"match": "^\\s*(set)\\s+(hi(?:s(?:t(?:o(?:r(?:y)?)?)?)?)?\\s+si(?:z(?:e)?)?)\\s+(unlimited|\\d+)\\s*$",
+					"captures": {
+						"1": { "name": "keyword.control.command.gdb" },
+						"2": { "name": "support.function.option.gdb" },
+						"3": { "name": "variable.other.variable.gdb" }
+					}
+				},
+				{
+					"name": "meta.set.history.filename.gdb",
+					"match": "^\\s*(set)\\s+(hi(?:s(?:t(?:o(?:r(?:y)?)?)?)?)?\\s+f(?:i(?:l(?:e(?:n(?:a(?:m(?:e)?)?)?)?)?)?)?)\\s+(.*)\\s*$",
+					"captures": {
+						"1": { "name": "keyword.control.command.gdb" },
+						"2": { "name": "support.function.option.gdb" },
+						"3": { "name": "markup.underline.link.filename.gdb" }
+					}
+				},
+				{
+					"name": "meta.set.history.removedups.gdb",
+					"match": "^\\s*(set)\\s+(hi(?:s(?:t(?:o(?:r(?:y)?)?)?)?)?\\s+remove-duplicates)\\s+(unlimited|\\d+)\\s*$",
 					"captures": {
 						"1": { "name": "keyword.control.command.gdb" },
 						"2": { "name": "support.function.option.gdb" },
 						"3": { "name": "variable.other.variable.gdb" }
 					}
 				},
+				{
+					"name": "meta.set.subs.gdb",
+					"match": "^\\s*(set)\\s+(su(?:b(?:s(?:t(?:i(?:t(?:u(?:t(?:e(?:-(?:p(?:a(?:t(?:h)?)?)?)?)?)?)?)?)?)?)?)?)?)?)?)\\s+(.*)\\s+(.*)\\s*$",
```

Comment:
There is one ) closing parenthesis more than necessary, making the parser fail. juste before \\s+(.*)\\s+(.*)\\s*$

Fix is `^\\s*(set)\\s+(su(?:b(?:s(?:t(?:i(?:t(?:u(?:t(?:e(?:-(?:p(?:a(?:t(?:h)?)?)?)?)?)?)?)?)?)?)?)?)?)?)?\\s+(.*)\\s+(.*)\\s*$`

Reply:
regex-online says that there's also a `)?` too much before that

Reply:
fixed